components/Modal/ModalWrapper.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { useState, ReactNode } from "react"
+import Container from "./components/Container"
+import styled from "styled-components"
+
+import Header from './components/Header'
+import Main from './components/Main'
+import Footer from './components/Footer'
+import { Modal, LeftPanelModal, RightPanelModal, BottomPanelModal } from "./variants/index"
+
+import { StyleContext } from "./common/contexts"
+
+const sizeToWidth = {
+  "small": "49rem",
+  "medium": "63rem",
+  "large": "91rem",
+  "x-large": "105rem"
+}
+
+export interface ModalWrapperProps {
+  /** Contents to be rendered in the header section */
+  header: { title: string, rightSection?: ReactNode },
+  /** Contents to be rendered on in the body section */
+  body: ReactNode,
+  /** Contents to be rendered on in the footer section */
+  footer: ReactNode,
```

Comment:
maybe we can have default header and footer and allow user to pass them as null

Reply:
I supose we are bringing in some restriction so that we can ensure consitency.

@arorayash thoughts about this? In our discussion earlier we had concluded that the header and footer are always going to be there...